extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/MethodNameParser.java

```diff
@@ -543,12 +554,7 @@ private List<ClassInfo> getSuperClassInfos(IndexView indexView, ClassInfo entity
                 mappedSuperClassInfoElements.add(superClass);
             }
 
-            if (superClassType.kind() == Kind.CLASS) {
-                superClassType = superClass.superClassType();
-            } else if (superClassType.kind() == Kind.PARAMETERIZED_TYPE) {
-                ParameterizedType parameterizedType = superClassType.asParameterizedType();
-                superClassType = parameterizedType.owner();
```

Comment:
This was my initial problem: `owner()` doesn't return anything helpful in case of e.g.:
```
    // class Foo<ID extends Serializable> { @EmbeddedId ID id; }
    // class Bar<ID extends SomeMoreSpecificBaseId> extends FOO<ID> { }
    // class Baz extends Bar<BazId> { }
```
If `id` is involved in a repo method, it needs to traverse to the top of the hierarchy, but as `owner()` doesn't return anything in this case, it stops at `Bar` and doesn't collect `Foo`.

I don't see what `owner()` ist supposed to solve here (and other `Kind`s shouldn't pop up here either).

Reply:
Btw, when rolling back this change for showcase purposes, you'll get an IT failure:
```
[ERROR] Errors:
[ERROR]   PhoneCallResourceTest.testFindByAreaCode » Runtime java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
        [error]: Build step io.quarkus.spring.data.deployment.SpringDataJPAProcessor#build threw an exception: io.quarkus.spring.data.deployment.UnableToParseMethodException: Entity io.quarkus.it.spring.data.jpa.PhoneCall does not contain a field named: idAreaCode. Offending method is 'findByIdAreaCode' of repository 'io.quarkus.it.spring.data.jpa.PhoneCallRepository'.
```